sdk/ml/azure-ai-ml/tests/component/unittests/test_command_component_entity.py

```diff
@@ -451,16 +451,11 @@ def test_primitive_output(self):
         assert actual_component_dict2 == expected_rest_component
 
     def test_component_code_asset_ignoring_pycache(self) -> None:
+        from azure.ai.ml.entities._component.code import ComponentIgnoreFile
+
         component_yaml = "./tests/test_configs/components/basic_component_code_local_path.yml"
         component = load_component(component_yaml)
-        code_folder = "./tests/test_configs/components/helloworld_components_with_env/"
-        with tempfile.TemporaryDirectory(dir=code_folder) as tmp_dir:
-            # create temp folder and write some files under pycache
-            tmp_pycache_folder = Path(tmp_dir) / "__pycache__"
-            tmp_pycache_folder.mkdir()
-            (tmp_pycache_folder / "a.pyc").touch()
-            (tmp_pycache_folder / "b.pyc").touch()
-            # resolve and check if pycache exists in code asset
-            with component._resolve_local_code() as code:
-                pycache_path = Path(component_yaml).parent / code.path / Path(tmp_dir).name / "__pycache__"
-                assert not pycache_path.exists()
+        # as we leverage ignore file, so only check the attribute and type here
+        with component._resolve_local_code() as code:
+            assert hasattr(code, "_ignore_file")
+            assert isinstance(getattr(code, "_ignore_file"), ComponentIgnoreFile)
```

Comment:
hmm... what about check `code._ignore_file.is_exclude` instead? It's more like a user interface

Reply:
yes, this kind of confuse me a lot yesterday, will try for that

Reply:
update one version, test for the behaviour of `code._ignore_file.is_file_excluded`